### PR TITLE
Create Shared conftest file

### LINKED ISSUE #80: Create shared conftest file
## Expected Behavior

There is shared conftest file that create fixture of all models to be used in the tests, the files is located under the main folder

## Current Behavior

Each model creating its own conftest file and many creation methods are duplicated 

## Acceptance Criteria

- [ ] All tests are passed
- [ ] There are no `@pytest.fixture` inside the test files

### PR DESCRIPTION
**Pull request by ErezCohenn**
### Implemented Changes

Create a shared `conftest.py` file 'hich contains a fixture of all models to be used in the tests,
And from now on, any model that is required to put information in a conftest file, will put in the shared conftest file

### Connected Issue/Story

Close #80 
Depends #64 

### Definition of Done
- [x] Redirect and unite current conftest files
- [x] Transfer all the `@pytest.fixture` content from the tests files to the conftets file
- [x] The all existing tests passed

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes
- [x] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]()
- [x] This PR does not require updating any documentation

### CODE REVIEW

File: conftest.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    return Order.create(dog_owner_id=DogOwner.objects.get(id=1),
+                        daycare_id=DayCare.objects.get(id=1),
+                        start_date=timezone.now(),
+                        end_date=timezone.now() + datetime.timedelta(days=3),
+                        price_per_day=100,
+                        )
+
+
+@pytest.fixture
+def dogowner_message_to_daycare1():
```
Comment (jeniaSakirko) [inline conftest.py:22]: maybe `dogowner1_message_to_daycare1`

File: conftest.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    return Message.create(author=AuthorOptions.DogOwner,
+                          dogowner_id=DogOwner.objects.get(pk=1),
+                          daycare_id=DayCare.objects.get(pk=1),
+                          text='Hello this is the test message1 from owner to day care')
+
+
+@pytest.fixture
+def daycare1_reply_to_dogonwer_message():
```
Comment (jeniaSakirko) [inline conftest.py:30]: maybe `daycare1_reply_to_dogonwer1_message `

File: conftest.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    return Message.create(author=AuthorOptions.DayCare,
+                          dogowner_id=DogOwner.objects.get(pk=1),
+                          daycare_id=DayCare.objects.get(pk=1),
+                          text='This is reply to first message from daycare to owner')
+
+
+@pytest.fixture
+def daycare2_message_to_dogowner():
```
Comment (jeniaSakirko) [inline conftest.py:38]: maybe `daycare2_message_to_dogowner1`

File: conftest.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    return Message.create(author=AuthorOptions.DayCare,
+                          dogowner_id=DogOwner.objects.get(pk=1),
+                          daycare_id=DayCare.objects.get(pk=2),
+                          text='Hello this new chat between daycare2 and dogowner')
+
+
+@pytest.fixture
+def daycare3_message_to_dogowner():
```
Comment (jeniaSakirko) [inline conftest.py:46]: maybe `daycare3_message_to_dogowner1`

File: conftest.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    return Image.create(url="../../static/images/daycare_image_test_01.jpeg", daycare_id=create_daycare_user)
+
+
+@pytest.fixture
+def create_image2(create_daycare_user):
+    return Image.create(url="../../static/images/daycare_image_test_02.jpeg", daycare_id=create_daycare_user)
+
+
+@pytest.fixture
+def daycare():
```
Comment (jeniaSakirko) [inline conftest.py:89]: we have the same creation multiple time, please combine
line 72 `def create_daycare_user():`

File: conftest.py [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    return DayCare.create(email="daycare_review@gmail.com", username="daycare_review", password="pass",
+                          name="reviewDayCare", description="This is test daycare for review", price_per_day=10,
+                          capacity=50, area="Merkaz", city="Tel-Aviv", address="The best review")
+
+
+@pytest.fixture
+def dogowner():
```
Comment (jeniaSakirko) [inline conftest.py:96]: we have the same creation multiple time, please combine
line 54 `def create_dog_owner_user():`

File: conftest.py [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+    return DogOwner.create(email="dogowner_review@gmail.com", username='dogowner_review', password='pass',
+                           dog_name='dogReview', first_name='firstNameReview', last_name='lastNameReview',
+                           phone_number=1234567890, dog_race='review',
+                           dog_picture_url='https://www.google.com/review.jpg', dog_age=1, dog_weight=1, dog_gender='M')
+
+
+@pytest.fixture
+def review_data():
```
Comment (jeniaSakirko) [inline conftest.py:104]: I suggest creating such fixture for DogOwner and DayCare so we won't use hard-coded values in the tests, See comments in the tests

File: dogowner/test_dog_owner.py [hunk 8]
```diff
@@ -43,140 +13,153 @@ def test_create_dog_owner(self, create_dog_owner_user):
...
         assert dogOwner_01.user not in User.objects.all()
 
     def test_duplicate_creation_dog_owner_user(self):
         with pytest.raises(ValidationError, match="Invalid username - username already exist"):
             for i in range(2):
-                DogOwner.create(email=EMAIL, username='SAME_USER_NAME_03', password=PASSWORD,
-                                dog_name=DOG_NAME, first_name=FIRST_NAME, last_name=LAST_NAME,
-                                phone_number=PHONE_NUMBER, dog_race=DOG_RACE, dog_picture_url=DOG_PICTURE_URL,
-                                dog_age=DOG_AGE, dog_weight=DOG_WEIGHT, dog_gender=DOG_GENDER
+                DogOwner.create(email='testuser@gmail.com', username='SAME_USER_NAME_03', password='testpassowrd',
```
Comment (jeniaSakirko) [inline dogowner/test_dog_owner.py:24]: hard coded values prone to bugs

**Review by jeniaSakirko — CHANGES_REQUESTED** [review]

File: conftest.py (hunk 1 above)
Comment (ErezCohenn) [reply conftest.py:22]: I agree, it was implicated by @Yuval-Vino in message app tests so I want to hear what do you think also

File: conftest.py (hunk 2 above)
Comment (ErezCohenn) [reply conftest.py:30]: same as above

File: conftest.py (hunk 3 above)
Comment (ErezCohenn) [reply conftest.py:38]: same as above

File: conftest.py (hunk 4 above)
Comment (ErezCohenn) [reply conftest.py:46]: same as above

File: conftest.py (hunk 5 above)
Comment (ErezCohenn) [reply conftest.py:89]: Nice catch! will be changed

File: conftest.py (hunk 6 above)
Comment (ErezCohenn) [reply conftest.py:96]: Nice catch! will be changed

File: conftest.py (hunk 1 above)
Comment (Yuval-Vino) [reply conftest.py:22]: all tests use a single dog owner, so I don't see the benefit of IDing the single dogowner, do you see it differently? @ErezCohenn  @jeniaSakirko

File: dogowner/test_dog_owner.py (hunk 8 above)
Comment (Omeramsc) [reply dogowner/test_dog_owner.py:24]: How so? in static data, you might do better with explicit data.  if we're afraid of bugs, we can test it.

File: conftest.py (hunk 1 above)
Comment (ErezCohenn) [reply conftest.py:22]: umm, if it's only one dog-owner I think it can remain as it is right now.

File: main/views.py [hunk 9]
```diff
@@ -2,18 +2,22 @@
...
+from daycare.models import DayCare
+from daycare.views import daycare_home
 
 
 def index(request):
     if request.user.is_authenticated:
-        return redirect(to='homepage')
-
+        if DayCare.objects.filter(user=request.user).exists():
+            return redirect(to='homepage')
```
Comment (Omeramsc) [inline main/views.py:12]: The change is good, but I don't see how it is related to this PR.

File: conftest.py (hunk 1 above)
Comment (Omeramsc) [reply conftest.py:22]: Please refrain from changing names in this PR when it is not mandatory. You can open a new PR for that once this one is merged.

File: conftest.py (hunk 2 above)
Comment (Omeramsc) [reply conftest.py:30]: Please refrain from changing names in this PR when it is not mandatory. You can open a new PR for that once this one is merged.

File: conftest.py (hunk 3 above)
Comment (Omeramsc) [reply conftest.py:38]: Please refrain from changing names in this PR when it is not mandatory. You can open a new PR for that once this one is merged.

File: conftest.py (hunk 4 above)
Comment (Omeramsc) [reply conftest.py:46]: Please refrain from changing names in this PR when it is not mandatory. You can open a new PR for that once this one is merged.

**Review by Omeramsc — APPROVED** [review]
LGTM but fix Jenia's comments.

**Comment by Omeramsc** [comment]
@jeniaSakirko @Yuval-Vino once Jenia gives his +1 on your changes, you can merge this. (make sure you're merging the dependency first)

**Review by jeniaSakirko — APPROVED** [review]
LGTM

**Pull request merged**